src/popup.js

```diff
@@ -169,10 +174,13 @@ function setupRoleFilter() {
 
 function sendSwitchRole(tabId, data) {
   executeAction(tabId, 'switch', data).then(() => {
-    let swcnt = localStorage.getItem('switchCount') || 0;
-    localStorage.setItem('switchCount', ++swcnt);
-    window.close()
+    localStorageRepo.get(['switchCount'])
+      .then(data => {
+        let swcnt = data.switchCount || 0;
+        localStorageRepo.set({ switchCount: ++swcnt }).then(() => {})
+      })
   });
+  window.close()
```

Comment:
Why did you move this line out of the promise chain?

Reply:
It is wrong solution so please revert last commit. This problem isn't related to manifest v3.

Reply:
It is fixed by #267 .

Reply:
The problem was caused by a different callback, sorry. I deleted the last commit.